ripple/src/main/webapp/script.js

```diff
@@ -1,37 +1,169 @@
+      for (var i = 0; i < it; i++) {
+        // append slides to end
+        if (e.direction == "left") {
+          $(".carousel-item")
+            .eq(i)
+            .appendTo(".carousel-inner");
+        } else {
+          $(".carousel-item")
+            .eq(0)
+            .appendTo($(this).find(".carousel-inner"));
+        }
+      }
+    }
+  });
+
+/* Code to dynamically load carousels. Calls addDynamicCarousel function to populate the carousels. */
+function loadCarousels() {
+    addDynamicCarousel("black-owned-businesses", "Black-owned");
+    addDynamicCarousel("under-10-mins-away", "5-10");
+    addDynamicCarousel("trending-near-you", "Trending");
+    addDynamicCarousel("up-and-coming", "New");
+}
+
+/* Dynamically loads content into Bootstrap carousel by reconstructing HTML elements and making a Firestore query. */
+function addDynamicCarousel(carouselId, tag) {
+    db.collection("businessClusters").get().then((querySnapshot) => {
+        var contentStrings = [];
+        var makeElement = false;
+        querySnapshot.forEach((doc) => {
+            //Check if the city in Firestore matches the city extracted from the user inputted address.
+            //Also checks if the business's tag list contains the tag that the carousel requires.
+            if (doc.data().city == localStorage.getItem('enteredCity') && doc.data().tags[tag] == true) {
+              makeElement = true;
+              const card = document.createElement('div');
+              card.classList = 'carousel-inner row w-100 mx-auto';
+
+            // Construct card content by appending HTML strings
+              var content = `
+                <div class="carousel-item col-md-4">
+                <div class="card">
+                    <img class="card-img-top img-fluid" id="card-dynamic-image" src = "/serve?blob-key=${doc.data().thumbnailImage}">
+                    <div class="card-body">
+                    <p class="card-text">${doc.data().name}</p>
+                    <p class="card-text"><small class="text-muted">${doc.data().walkingDistance}</small></p>
+                    </div>
+                </div>
+                </div>
+              `;
+              //Add every line of reconstructed HTML to contentStrings array
+              contentStrings.push(content);
+            }
+        });
+        // Append newly created card element to the container
+        if (makeElement) {
+          var carouselElement = document.getElementById(carouselId);
+          var carouselInner = carouselElement.getElementsByClassName('carousel-inner row w-100 mx-auto')[0];
+          alert(contentStrings.join("\n"));
+          carouselInner.innerHTML = contentStrings.join("\n");
+          alert(document.getElementById('card-dynamic-image').src);
+          carouselInner.firstElementChild.className += " active";
+          $(carouselElement).carousel({slide : true, interval : false});
+        }
+   });
+}
+
```

Comment:
Add landing page-specific and main page-specific functions either to landing.js or main.js logic. Script.js should contain functions that every page will use.

Reply:
good point, has been updated